### PR TITLE
Throttle onChange in text inputs

### PR DESCRIPTION
**Pull request by skogsmaskin**
### Description

This will introduce a local editing state on String and Text inputs and then submits the produced patches throttled (1000 ms throttle time, non-leading).

This makes the performance and writing experience in these fields a lot better.

### Notes for release
* Better typing performance in String and Text inputs.

### CODE REVIEW

File: packages/@sanity/form-builder/src/inputs/StringInput.tsx [hunk 1]
```diff
@@ -19,29 +20,60 @@ const StringInput = React.forwardRef(function StringInput(
     [markers]
   )
 
-  const handleChange = React.useCallback(
+  const [localValue, setlocalValue] = useState(null)
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/StringInput.tsx:23]: nit: `setlocalValue` => `setLocalValue`

File: packages/@sanity/form-builder/src/inputs/StringInput.tsx [hunk 2]
```diff
@@ -19,29 +20,60 @@ const StringInput = React.forwardRef(function StringInput(
...
       const inputValue = event.currentTarget.value
-      onChange(PatchEvent.from(inputValue ? set(inputValue) : unset()))
+      setlocalValue(inputValue)
+      throttledSubmit()
     },
-    [onChange]
+    [throttledSubmit]
   )
 
+  useEffect(() => () => throttledSubmit(), [throttledSubmit])
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/StringInput.tsx:49]: should be `throttledSubmit.flush()` here I think?

File: packages/@sanity/form-builder/src/inputs/TextInput.tsx [hunk 3]
```diff
@@ -26,13 +27,34 @@ const TextInput = React.forwardRef(function TextInput(
     [markers]
   )
 
-  const handleChange = React.useCallback(
-    (event) => {
+  const [localValue, setlocalValue] = useState(null)
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/TextInput.tsx:30]: nit: `setlocalValue` => `setLocalValue`

File: packages/@sanity/form-builder/src/inputs/TextInput.tsx [hunk 4]
```diff
@@ -26,13 +27,34 @@ const TextInput = React.forwardRef(function TextInput(
...
       const inputValue = event.currentTarget.value
-      onChange(PatchEvent.from(inputValue ? set(inputValue) : unset()))
+      setlocalValue(inputValue)
+      throttledSubmit()
     },
-    [onChange]
+    [throttledSubmit]
   )
+
+  useEffect(() => () => throttledSubmit(), [throttledSubmit])
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/TextInput.tsx:56]: Same as above, should be `throttledSubmit.flush()` here as well I believe

**Review by bjoerge — CHANGES_REQUESTED** [review]

**Comment by bjoerge** [comment]
Seems to work fine! Just a couple of nitpicks, and a comment about calling `.flush()` that would be good if you could look into :)

File: packages/@sanity/form-builder/src/inputs/StringInput.tsx (hunk 2 above)
Comment (skogsmaskin) [reply packages/@sanity/form-builder/src/inputs/StringInput.tsx:49]: Yes! I was actually thinking about this in the weekend. My idea was just to call onChange there directly.

File: packages/@sanity/form-builder/src/inputs/TextInput.tsx (hunk 4 above)
Comment (skogsmaskin) [reply packages/@sanity/form-builder/src/inputs/TextInput.tsx:56]: Yeah

**Pull request merged**